### DISCUSSION TITLE
Warnings for Controller `defaultValue` or useForm `defaultValues` is missing after upgrading to 6.4 from 6.0.1

### DISCUSSION THREAD

**Discussion by ajaybeniwal** [react-hook-form/react-hook-form#2656] [Q&A] [answered]
We decided to upgrade our React-Hooks-Form Library and we started using getting these messages 
Controller `defaultValue` or useForm `defaultValues` is missing , we are using Controller with Antd Controls
```
const form = useForm({
		mode: 'all',
		defaultValues: {},
	});
```

```
<Controller
			as={
				<FormInput
					layout={formItemLayout || (isTextArea ? formItemLayoutTextArea : formItemLayoutInput)}
					label={label}
					validateStatus={!isEmpty(get(errors, `${id}`)) ? 'error' : undefined}
					help={get(errors, `${id}.message`)}
					seleniumId={seleniumId}
					messages={messages}
					rows={4}
					hasCount={hasCount}
					isTextArea={isTextArea}
					max={max}
					disabled={disabled}
					colon={colon}
					addOnBefore={addOnBefore}
				/>
			}
			name={id}
			control={control}
```

**Comment by bluebill1049** [comment, marked as answer]
you should give Controller a defaultValue: either at useForm or Controller

eg:
```
useForm({
   defaultValues: {
    test: ''
  }
})
```

**Reply by ajaybeniwal** [reply]
Okay i have two question 
1. Will it cause any performance issues or logical issues if we do not provide default values and will it also display console.warn in production build as well
2. Our form is very dynamic and big form and when user lands on the page none of the default values will be selected so how do you suggest we tackle this problem as we want to avoid defining the shape of form in defaultValues

**Reply by bluebill1049** [reply]
1. nope (remove in prod build)
2. still you should have defaultValue

**Comment by areeb-hub** [comment]
@bluebill1049 I've given defaultValues for useForm({}) still it shows the same warning.

**Reply by bluebill1049** [reply]
```
useForm({
  defaultValues: xxx // here
})
```

**Comment by TOcvfan** [comment]
I have 
`useForm({
   defaultValues: {
    km_slut: ''
  }
})`

and 
```
<Controller
              control={control}
              name="km_slut"
              defaultValue={kmSlut}
              render={(props) => (
                <TextField
                  variant="outlined"
                  placeholder="Skriv slut kilometer her"
                  onChange={(e) => props.onChange(setKmSlut(e.target.value))}
                  defaultValue={kmSlut}
                />        
              )}
          />
```

but I still get 'Controller `defaultValue` or useForm `defaultValues` is missing'

**Reply by bluebill1049** [reply]
can you reproduce that in a codesandbox?

**Reply by TOcvfan** [reply]
yes here [https://codesandbox.io/s/happy-benz-zb6t6?file=/src/App.js](url)

**Reply by bluebill1049** [reply]
you are doing it wrong, take a look at this codesandbox: fork it https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by TOcvfan** [reply]
you mean like this [https://codesandbox.io/s/happy-benz-zb6t6?file=/src/InputTextField.js](url) but how do I get the values from it to my hooks, so I can calculate the kilometers?

**Reply by bluebill1049** [reply]
setValue api

**Reply by TOcvfan** [reply]
yes I have seen it and also getValues, but how do I call it without having to push a button, so it gets calculated as they are typing?

**Reply by bluebill1049** [reply]
so you want to get the value of the inputs? `watch` then

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api#watch

**Reply by TOcvfan** [reply]
thanks watch work I just had to use the little gray ones :-)
